_posts/2021-03-21-clean-unit-tests-with-mockito.md

```diff
@@ -0,0 +1,681 @@
+Mockito.verify(cityRepository, Mockito.times(1)).delete(expected);
+```
+
+We can verify how many times a mock was called by simply use the built-in `verify` method. If the condition is not met,
+our test case will fail. This is extremely handy for algorithms or similar processes. There are other predefined
+verification modes such as `atLeastOnce` or `never` already present and ready to use!
+
+## Mockito FAQ
+
+In this section we want to point out important things which are nice to know.
+
+* *What types can I mock?* Mockito allows us to mock not only interfaces but also concrete classes.
+* *What is returned if I don't declare a mock's behaviour?* Mockito returns `null` for reference objects, and the
+  default values for primitive data types (for example `0` for `int` and `false` for `boolean`)
+* *How many times does Mockito return a previously declared value?* Mockito returns always the same value, regardless of
+  how many times a method is called.
+* *Can I mock `final` classes?* No, final classes **can't** be mocked and neither final methods are mockable. This has
+  to do with the internal mechanism of how Mocktio creates the mock and the Java Language Specification. If we want to
+  do so, we use [PowerMock](https://github.com/powermock/powermock).
+* *Can I mock a constructor?* Mockito can't mock constructors, static methods, `equals()` nor `hashCode` out of the box.
+  In order to achieve that [PowerMock](https://github.com/powermock/powermock) must be used.
+
+## Pros and Cons
+
+Mockito helps us to create simple mocks fast. The application of the methods is easy to read, since they are written in
+fluent style. Mockito can be used in plain Java projects or together with frameworks such as Spring Boot. It Is well
+documented and has lots of examples in it. In case of problems there is a huge community behind and questions are
+answered frequently on StackOverflow. It disposes great flexibility to its users which can contribute their ideas, since
+it is an open source project. Therefore, the development is ongoing, and the project is maintained.
+
+Mockito can't mock everything out of the box. In case we want to mock `final` or `static` methods, `equals()` or the
+construction of an object, we need [PowerMock](https://github.com/powermock/powermock).
+
+## Conclusion
+
+In this post we learned how to create mocks for unit tests in various variants. Mockito gives us a lot of flexibility,
+and the freedom to choose between numerous tools to achieve our goals. When working in teams, we define a common
+language and Mockito code style guideline how we want to use this powerful tool for testing. This will improve our
+performance and helps to discuss and communicate.
+
+Although Mockito comes with a lot of features, be aware of its restrictions. Don't spend time to make the impossible
+possible, better reconsider our approach to test a scenario.
+
+You will find all examples on [GitHub](https://github.com/silenum/mockito-examples).
```

Comment:
I would like all code examples to be available in the [code examples](https://github.com/thombergs/code-examples) repository. Could you create a PR to add that module in that repo with the code?